### ISSUE TITLE
Selecting SURFnet bv and selecting Norway causes traceback

### ISSUE THREAD

**Issue by amarsman** [eduvpn/python-eduvpn-client#329]
Started with a clean ~/.config/eduvpn

# Installation

git

# Version

Top commit:

ommit 1d03ec5703da35be3da824812d67c44073bbdc59 (HEAD -> 2.0.x, origin/2.0.x)
Author: Alexander van Ratingen <470642+alvra@users.noreply.github.com>
Date:   Thu Mar 4 18:22:45 2021 +0100

    Handle errors on getting server info.

# OS/Distribution

[Ubuntu](url) 20.04

# Logs

Putting child 0x55cf84ddeff0 (eduvpn-gui) PID 5226 on the chain.
Live child 0x55cf84ddeff0 (eduvpn-gui) PID 5226 

(eduvpn-gui:5226): Gtk-WARNING **: 10:50:44.780: Overriding tab label for notebook
2021-03-05 10:50:44,785 - MainThread - INFO - eduvpn.ui.ui - ui.py:121 - starting ui
2021-03-05 10:50:44,791 - init-server-db - INFO - eduvpn.remote - remote.py:17 - Requesting https://disco.eduvpn.org/v2/server_list.json
2021-03-05 10:50:44,816 - MainThread - INFO - eduvpn.app - app.py:103 - network transitioning: <InitialNetworkState> -> no_previous_connection_found
2021-03-05 10:50:44,816 - MainThread - INFO - eduvpn.app - app.py:122 - interface transitioning: <InitialInterfaceState> -> no_active_connection_found
2021-03-05 10:50:44,820 - MainThread - INFO - eduvpn.app - app.py:133 - interface transitioned: no_active_connection_found -> <ConfigurePredefinedServer search_query='' results=None>
2021-03-05 10:50:44,820 - MainThread - INFO - eduvpn.app - app.py:114 - network transitioned: no_previous_connection_found -> <UnconnectedState>
2021-03-05 10:50:45,029 - init-server-db - INFO - eduvpn.remote - remote.py:31 - Retrieving signature https://disco.eduvpn.org/v2/server_list.json.minisig
2021-03-05 10:50:45,161 - init-server-db - INFO - eduvpn.remote - remote.py:38 - verifying signature of https://disco.eduvpn.org/v2/server_list.json
2021-03-05 10:50:45,162 - init-server-db - INFO - eduvpn.remote - remote.py:17 - Requesting https://disco.eduvpn.org/v2/organization_list.json
2021-03-05 10:50:45,414 - init-server-db - INFO - eduvpn.remote - remote.py:31 - Retrieving signature https://disco.eduvpn.org/v2/organization_list.json.minisig
2021-03-05 10:50:45,548 - init-server-db - INFO - eduvpn.remote - remote.py:38 - verifying signature of https://disco.eduvpn.org/v2/organization_list.json
2021-03-05 10:50:45,553 - MainThread - INFO - eduvpn.app - app.py:122 - interface transitioning: <ConfigurePredefinedServer search_query='' results=None> -> server_db_finished_loading
2021-03-05 10:50:45,553 - MainThread - INFO - eduvpn.app - app.py:133 - interface transitioned: server_db_finished_loading -> <ConfigurePredefinedServer search_query='' results=None>
2021-03-05 10:50:46,937 - MainThread - INFO - eduvpn.app - app.py:122 - interface transitioning: <ConfigurePredefinedServer search_query='' results=None> -> enter_search_query
2021-03-05 10:50:46,944 - MainThread - INFO - eduvpn.app - app.py:133 - interface transitioned: enter_search_query -> <ConfigurePredefinedServer search_query='sur' results=4>
2021-03-05 10:50:47,128 - MainThread - INFO - eduvpn.app - app.py:122 - interface transitioning: <ConfigurePredefinedServer search_query='sur' results=4> -> enter_search_query
2021-03-05 10:50:47,132 - MainThread - INFO - eduvpn.app - app.py:133 - interface transitioned: enter_search_query -> <ConfigurePredefinedServer search_query='surf' results=3>
2021-03-05 10:50:49,098 - MainThread - INFO - eduvpn.ui.ui - ui.py:403 - selection empty
2021-03-05 10:50:49,098 - MainThread - INFO - eduvpn.app - app.py:122 - interface transitioning: <ConfigurePredefinedServer search_query='surf' results=3> -> connect_to_server
2021-03-05 10:50:49,099 - MainThread - INFO - eduvpn.remote - remote.py:17 - Requesting https://nl.eduvpn.org/info.json
2021-03-05 10:50:49,130 - MainThread - INFO - eduvpn.app - app.py:122 - interface transitioning: <ConfigurePredefinedServer search_query='surf' results=3> -> enter_search_query
2021-03-05 10:50:49,131 - MainThread - INFO - eduvpn.app - app.py:133 - interface transitioned: enter_search_query -> <ConfigurePredefinedServer search_query='' results=None>
2021-03-05 10:50:49,131 - MainThread - INFO - eduvpn.app - app.py:133 - interface transitioned: connect_to_server -> <OAuthSetup server=<OrganisationServer 'SURFnet bv'>, oauth_web_server=<eduvpn.oauth2.OAuthWebServer object at 0x7feca8dcad60>>
2021-03-05 10:50:49,133 - oauth-http-server - INFO - eduvpn.oauth2 - oauth2.py:125 - opening browser with https://nl.eduvpn.org/portal/_oauth/authorize?response_type=code&client_id=org.eduvpn.app.linux&redirect_uri=http%3A%2F%2F127.0.0.1%3A47759%2Fcallback&scope=config&state=AukepQeQqYuXhFbpxdtUBWhwKkbGzj&code_challenge_method=S256&code_challenge=TFlsyuyzROBX0n_EOC20HgnwXDmVP0IgGmMyZx071g8
2021-03-05 10:50:49,369 - oauth-http-server - INFO - eduvpn.oauth2 - oauth2.py:78 - listening for a request on port 47759...
127.0.0.1 - - [05/Mar/2021 10:50:54] "GET /callback?code=eyJhbGciOiJFZERTQSIsInR5cCI6IkpXVCIsImtpZCI6InhHQXhvNnhTOVIzQ0hYY19mWWh6ZVlBQ29CMWRUSENlbjFtWEVkLXZtaEUifQ.eyJ2Ijo1LCJ0eXBlIjoiYXV0aG9yaXphdGlvbl9jb2RlIiwiYXV0aF9rZXkiOiIzMTFkY2ViMzM1ZmNkODk0ZWViYzBlZWJmZDM1MGU4NiIsInVzZXJfaWQiOiJodHRwczpcL1wvZXZhLXNhbWwtaWRwLmVkdXJvYW0ubmxcL3NpbXBsZXNhbWxwaHBcL3NhbWwyXC9pZHBcL21ldGFkYXRhLnBocCFodHRwczpcL1wvbmwuZWR1dnBuLm9yZ1wvc2FtbCE3OTg3M2U5Nzk4NjFlYWQxZWVkZWMzNjljMjMxYjQ4NTI2ODI3NjIyIiwiY2xpZW50X2lkIjoib3JnLmVkdXZwbi5hcHAubGludXgiLCJzY29wZSI6ImNvbmZpZyIsInJlZGlyZWN0X3VyaSI6Imh0dHA6XC9cLzEyNy4wLjAuMTo0Nzc1OVwvY2FsbGJhY2siLCJjb2RlX2NoYWxsZW5nZSI6IlRGbHN5dXl6Uk9CWDBuX0VPQzIwSGdud1hEbVZQMElnR21NeVp4MDcxZzgiLCJleHBpcmVzX2F0IjoiMjAyMS0wMy0wNVQwOTo1NTo1NCswMDowMCJ9.y6F3wMO5RCrUeAXDoPHjDPpv8-s_oNbKeZND54XS_Inyx-K1SnDOp_AOjz7VjfBh8D74LmLCk8fRaWfiszrEBA&state=AukepQeQqYuXhFbpxdtUBWhwKkbGzj HTTP/1.1" 200 -
2021-03-05 10:50:54,980 - oauth-http-server - INFO - eduvpn.oauth2 - oauth2.py:106 - received a request /callback?code=eyJhbGciOiJFZERTQSIsInR5cCI6IkpXVCIsImtpZCI6InhHQXhvNnhTOVIzQ0hYY19mWWh6ZVlBQ29CMWRUSENlbjFtWEVkLXZtaEUifQ.eyJ2Ijo1LCJ0eXBlIjoiYXV0aG9yaXphdGlvbl9jb2RlIiwiYXV0aF9rZXkiOiIzMTFkY2ViMzM1ZmNkODk0ZWViYzBlZWJmZDM1MGU4NiIsInVzZXJfaWQiOiJodHRwczpcL1wvZXZhLXNhbWwtaWRwLmVkdXJvYW0ubmxcL3NpbXBsZXNhbWxwaHBcL3NhbWwyXC9pZHBcL21ldGFkYXRhLnBocCFodHRwczpcL1wvbmwuZWR1dnBuLm9yZ1wvc2FtbCE3OTg3M2U5Nzk4NjFlYWQxZWVkZWMzNjljMjMxYjQ4NTI2ODI3NjIyIiwiY2xpZW50X2lkIjoib3JnLmVkdXZwbi5hcHAubGludXgiLCJzY29wZSI6ImNvbmZpZyIsInJlZGlyZWN0X3VyaSI6Imh0dHA6XC9cLzEyNy4wLjAuMTo0Nzc1OVwvY2FsbGJhY2siLCJjb2RlX2NoYWxsZW5nZSI6IlRGbHN5dXl6Uk9CWDBuX0VPQzIwSGdud1hEbVZQMElnR21NeVp4MDcxZzgiLCJleHBpcmVzX2F0IjoiMjAyMS0wMy0wNVQwOTo1NTo1NCswMDowMCJ9.y6F3wMO5RCrUeAXDoPHjDPpv8-s_oNbKeZND54XS_Inyx-K1SnDOp_AOjz7VjfBh8D74LmLCk8fRaWfiszrEBA&state=AukepQeQqYuXhFbpxdtUBWhwKkbGzj
2021-03-05 10:50:55,037 - MainThread - INFO - eduvpn.app - app.py:122 - interface transitioning: <OAuthSetup server=<OrganisationServer 'SURFnet bv'>, oauth_web_server=<eduvpn.oauth2.OAuthWebServer object at 0x7feca8dcad60>> -> oauth_setup_success
2021-03-05 10:50:55,037 - oauth-refresh - INFO - eduvpn.remote - remote.py:17 - Requesting https://nl.eduvpn.org/info.json
2021-03-05 10:50:55,038 - MainThread - INFO - eduvpn.app - app.py:133 - interface transitioned: oauth_setup_success -> <OAuthRefreshToken app=<eduvpn.app.Application object at 0x7fecb2e8fac0>, server=<OrganisationServer 'SURFnet bv'>, oauth_session=<requests_oauthlib.oauth2_session.OAuth2Session object at 0x7feca8dcadf0>>
2021-03-05 10:50:55,082 - MainThread - INFO - eduvpn.app - app.py:122 - interface transitioning: <OAuthRefreshToken app=<eduvpn.app.Application object at 0x7fecb2e8fac0>, server=<OrganisationServer 'SURFnet bv'>, oauth_session=<requests_oauthlib.oauth2_session.OAuth2Session object at 0x7feca8dcadf0>> -> oauth_refresh_success
2021-03-05 10:50:55,082 - MainThread - INFO - eduvpn.remote - remote.py:17 - Requesting https://nl.eduvpn.org/info.json
2021-03-05 10:50:55,134 - MainThread - INFO - eduvpn.app - app.py:133 - interface transitioned: oauth_refresh_success -> <ChooseSecureInternetLocation server=<OrganisationServer 'SURFnet bv'>, oauth_session=<requests_oauthlib.oauth2_session.OAuth2Session object at 0x7feca8dcadf0>, locations=[<SecureInternetServer 'Albania @ https://eduvpn.rash.al/'>, <SecureInternetServer 'Australia @ https://gdpt-eduvpndev1.tnd.aarnet.edu.au/'>, <SecureInternetServer 'Denmark @ https://eduvpn.deic.dk/'>, <SecureInternetServer 'Estonia @ https://eduvpn.eenet.ee/'>, <SecureInternetServer 'Finland @ https://eduvpn1.funet.fi/'>, <SecureInternetServer 'France @ https://eduvpn-poc.renater.fr/'>, <SecureInternetServer 'Germany @ https://eduvpn1.eduvpn.de/'>, <SecureInternetServer 'Malaysia @ https://eduvpn.myren.net.my/'>, <SecureInternetServer 'Morocco @ https://eduvpn.marwan.ma/'>, <SecureInternetServer 'Norway @ https://guest.eduvpn.no/'>, <SecureInternetServer 'Pakistan @ https://vpn.pern.edu.pk/'>, <SecureInternetServer 'South Africa @ https://guest.eduvpn.ac.za/'>, <SecureInternetServer 'Sri Lanka @ https://eduvpn.ac.lk/'>, <SecureInternetServer 'The Netherlands @ https://nl.eduvpn.org/'>, <SecureInternetServer 'Uganda @ https://eduvpn.renu.ac.ug/'>, <SecureInternetServer 'Ukraine @ https://eduvpn.uran.ua/'>]>
2021-03-05 10:50:57,046 - MainThread - INFO - eduvpn.ui.ui - ui.py:454 - selection empty
2021-03-05 10:50:57,046 - MainThread - INFO - eduvpn.app - app.py:122 - interface transitioning: <ChooseSecureInternetLocation server=<OrganisationServer 'SURFnet bv'>, oauth_session=<requests_oauthlib.oauth2_session.OAuth2Session object at 0x7feca8dcadf0>, locations=[<SecureInternetServer 'Albania @ https://eduvpn.rash.al/'>, <SecureInternetServer 'Australia @ https://gdpt-eduvpndev1.tnd.aarnet.edu.au/'>, <SecureInternetServer 'Denmark @ https://eduvpn.deic.dk/'>, <SecureInternetServer 'Estonia @ https://eduvpn.eenet.ee/'>, <SecureInternetServer 'Finland @ https://eduvpn1.funet.fi/'>, <SecureInternetServer 'France @ https://eduvpn-poc.renater.fr/'>, <SecureInternetServer 'Germany @ https://eduvpn1.eduvpn.de/'>, <SecureInternetServer 'Malaysia @ https://eduvpn.myren.net.my/'>, <SecureInternetServer 'Morocco @ https://eduvpn.marwan.ma/'>, <SecureInternetServer 'Norway @ https://guest.eduvpn.no/'>, <SecureInternetServer 'Pakistan @ https://vpn.pern.edu.pk/'>, <SecureInternetServer 'South Africa @ https://guest.eduvpn.ac.za/'>, <SecureInternetServer 'Sri Lanka @ https://eduvpn.ac.lk/'>, <SecureInternetServer 'The Netherlands @ https://nl.eduvpn.org/'>, <SecureInternetServer 'Uganda @ https://eduvpn.renu.ac.ug/'>, <SecureInternetServer 'Ukraine @ https://eduvpn.uran.ua/'>]> -> select_secure_internet_location
2021-03-05 10:50:57,046 - MainThread - INFO - eduvpn.remote - remote.py:17 - Requesting https://nl.eduvpn.org/info.json
2021-03-05 10:50:57,083 - MainThread - INFO - eduvpn.remote - remote.py:17 - Requesting https://guest.eduvpn.no/info.json
2021-03-05 10:50:57,624 - MainThread - INFO - eduvpn.remote - remote.py:17 - Requesting https://nl.eduvpn.org/info.json
2021-03-05 10:50:57,662 - MainThread - INFO - eduvpn.remote - remote.py:17 - Requesting https://nl.eduvpn.org/info.json
2021-03-05 10:50:57,702 - MainThread - INFO - eduvpn.app - app.py:133 - interface transitioned: select_secure_internet_location -> <ConfiguringConnection server=<SecureInternetLocation <OrganisationServer 'SURFnet bv'> <SecureInternetServer 'Norway @ https://guest.eduvpn.no/'>>>
2021-03-05 10:51:00,197 - MainThread - INFO - eduvpn.nm - nm.py:129 - writing configuration to Network Manager
2021-03-05 10:51:00,198 - MainThread - INFO - eduvpn.storage - storage.py:162 - Writing configuration to /tmp/tmp3l2zq4pp/eduVPN.ovpn
Traceback (most recent call last):
  File "/media/psf/Home/projects/eduvpn/python-eduvpn-client/alvra/python-eduvpn-client/eduvpn/interface/event.py", line 216, in save_connection
    nm.save_connection(nm.get_client(), config, private_key, certificate,
  File "/media/psf/Home/projects/eduvpn/python-eduvpn-client/alvra/python-eduvpn-client/eduvpn/nm.py", line 130, in save_connection
    new_con = import_ovpn(config, private_key, certificate)
  File "/media/psf/Home/projects/eduvpn/python-eduvpn-client/alvra/python-eduvpn-client/eduvpn/nm.py", line 91, in import_ovpn
    connection = nm_ovpn_import(target)
  File "/media/psf/Home/projects/eduvpn/python-eduvpn-client/alvra/python-eduvpn-client/eduvpn/nm.py", line 79, in nm_ovpn_import
    conn = vpn_infos[0].load_editor_plugin().import_(str(target))
gi.repository.GLib.Error: nm-connection-error-quark: The file to import wasn’t a valid OpenVPN client configuration (0)
[eduVPN.ovpn.txt](https://github.com/eduvpn/python-eduvpn-client/files/6089691/eduVPN.ovpn.txt)

**Comment by alvra** [comment]
Thanks for the report.
The ovpn file looks suspicious, however I'm not able to reproduce this on Ubuntu 20.04.1. Which server and profile are you connecting to?

**Comment by amarsman** [comment]
It's in the title and in the provided logging: SURFnet bv and selecting Norway.

**Comment by alvra** [comment]
Sorry I missed that part. I'm seeing it too for Norway.

**Comment by alvra** [comment]
Looks like there's a problem getting the profile for this specific combination. I can trace the error back to: https://nl.eduvpn.org/portal/api.php/profile_config?profile_id=internet
which returns:
```
{"profile_config":{"ok":false,"error":"profile not available or no permission"}}
```
I'll update the app to check the response, but it seems like there's something going wrong en their end.

**Comment by amarsman** [comment]
I tried my last version and there SURFnet bv and selecting Norway works without an error so to me it doesn't look like a server side problem.

**Comment by amarsman** [comment]
Retested on master.

Reopening this issue as it still fails to connect to SURFnet bv and selecting Norway on master while that works with my branch: https://github.com/amarsman/python-eduvpn-client/tree/2.0.x-issues

**Comment by alvra** [comment]
This issue has been fixed in #338.
Thanks for reporting this @amarsman
